include/net/rest_client.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file rest_client.h
+ *
+ * @defgroup rest_client REST client library
+ * @{
+ * @brief REST client.
+ *
+ * @details Provide REST client functionality.
+ */
+
+#ifndef REST_CLIENT_H__
+#define REST_CLIENT_H__
+
+#include <net/http_parser.h>
+
+/** @brief TLS is not used. */
+#define REST_CLIENT_SEC_TAG_NO_SEC -1
+
+/** @brief Use the default TLS peer verification; TLS_PEER_VERIFY_REQUIRED. */
+#define REST_CLIENT_TLS_DEFAULT_PEER_VERIFY -1
+
+/** @brief REST client opens a socket connection. */
+#define REST_CLIENT_SCKT_CONNECT -1
+
+/** @brief Common HTTP status codes */
+enum rest_client_http_status {
+	REST_CLIENT_HTTP_STATUS_OK = 200,
+	REST_CLIENT_HTTP_STATUS_BAD_REQ = 400,
+	REST_CLIENT_HTTP_STATUS_UNAUTH = 401,
+	REST_CLIENT_HTTP_STATUS_FORBIDDEN = 403,
+	REST_CLIENT_HTTP_STATUS_NOT_FOUND = 404,
+};
+
+/**
+ * @brief REST client request and respoonse contexts.
+ *
+ * @details Input parameters for using the REST client API should be filled into
+ *          @ref rest_client_resp_context before calling @ref rest_client_request
+ *          and when the function returns, response-related data can be read from
+ *          @ref rest_client_resp_context structure.
+ */
```

Comment:
I think it would be better to have own descriptions for both `rest_client_req_context` and `rest_client_resp_context`. And "respoonse" -> "response". :)

Reply:
Split the comments to req and resp